src/main/frontend/util/cursor.cljs

```diff
@@ -0,0 +1,195 @@
+(ns frontend.util.cursor
+  (:require [cljs-bean.core :as bean]
+            [clojure.string :as string]
+            [frontend.util :as util]
+            [goog.dom :as gdom]
+            [goog.object :as gobj]))
+
+(defn- closer [a b c]
+  (let [a-left (or (:left a) 0)
+        b-left (:left b)
+        c-left (or (:left c) js/Number.MAX_SAFE_INTEGER)]
+    (if (< (- b-left a-left) (- c-left b-left))
+      a
+      c)))
+
+(defn mock-char-pos [e]
+  {:left (.-offsetLeft e)
+   :top  (.-offsetTop e)
+   :pos  (-> (.-id e)
+             (string/split "_")
+             second
+             int)})
+
+(defn get-caret-pos
+  [input]
+  (let [pos (.-selectionStart input)
+        rect (bean/->clj (.. input (getBoundingClientRect) (toJSON)))]
+    (try
+      (-> (gdom/getElement "mock-text")
+          gdom/getChildren
+          array-seq
+          (nth pos)
+          mock-char-pos
+          (assoc :rect rect))
+      (catch :default _e
```

Comment:
just a safeguard for nth fn, for now, no situation i'm aware of would trigger this.

but lets say if someone added some listener on single appended char with autocomplete like `/ ` and need to call this caret pos immediately, now the correct place to do so is in the mixin of mock-textarea, otherwise this would happen.

Reply:
cool.